src/main/java/de/taimos/gpsd4java/backend/SocketThread.java

```diff
@@ -114,15 +113,14 @@ protected void retry() {
 	/**
 	 * Halts the socket thread.
 	 *
-	 * @throws InterruptedException
 	 */
-	public void halt() throws InterruptedException {
+	public void halt() {
 		this.running.set(false);
+		
 		try {
 			this.reader.close();
 		} catch (final IOException e) {
 			// ignore
 		}
-		this.join(1000);
```

Comment:
As far as I understand, `Thread::join(long timeout)` doesn't guarantee the thread will have stopped by the time the timeout expires. IMHO, it seems kinda undeterministic whether the thread has finished when `SocketThread::halt()` (and consequently `GPSdEndpoint::stop()`) returns. 

Should we make the current running SocketThread accessible with a getter in  GPSdEndpoint, so that developers can better manage it and control their application's lifecycle?